test/unit/ensure_not_used_by_test.rb

```diff
@@ -0,0 +1,40 @@
+require 'test_helper'
+
+class EnsureNotUsedByTest < ActiveSupport::TestCase
+  def setup
+    @org1 = FactoryGirl.create(:organization)
+    @org2 = FactoryGirl.create(:organization)
+    @user = User.create :auth_source => auth_sources(:one), :login => "foo", :mail  => "foo@bar.com"
+    @user.organizations = [@org1]
+  end
+
+  test "hostgroup should not be deleted if used by host in user org" do
+    hostgroup = FactoryGirl.create(:hostgroup, :organizations => [@org1, @org2])
+    FactoryGirl.create(:host, :hostgroup => hostgroup, :organization => @org1)
+
+    as_user @user do
+      hostgroup.destroy
+      assert_present Hostgroup.where(:id => hostgroup.id)
+    end
+  end
+
+  test "hostgroup should not be deleted if used by host in different org" do
+    hostgroup = FactoryGirl.create(:hostgroup, :organizations => [@org1, @org2])
+    FactoryGirl.create(:host, :hostgroup => hostgroup, :organization => @org2)
+
+    as_user @user do
```

Comment:
I added `in_taxonomy @org1` in both tests since that is the user's org. For the first test Host.all should find the host but in the second one it won't be in the results.